barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.hpp

```diff
@@ -7,6 +7,111 @@
 
 namespace bb {
 
+/*! \brief The implementation of the sum-check Prover for the statements of the form \f$\sum_{\vec \ell \in \{0,1\}^d}
+pow_{\beta}(\vec \ell) \cdot F \left(P_1(\vec \ell),\ldots, P_N(\vec \ell) \right)  = 0 \f$ for multi-linear polynomials
+\f$P_1, \ldots, P_N \f$.
+
+   \details
+ ## Notation and Setup
+
+ ### Obtaining Prover/Honk Polynomials
+ The Sumcheck is applied to a \p Flavor that in its turn specifies the information about the multi-variate polynomials
+\f$P_1, \ldots, P_N\f$. Namely, \ref prove "prove method" obtains \p full_polynomials by reference from \p Flavor 's
+\ref ProverPolynomials "prover polynomials". In particular, their number \f$N\f$ is specified by the \p Flavor.
+
+ ### Sumcheck Relation
+ Given multi-linear polynomials \f$ P_1,\ldots, P_N \in \mathbb{F}[X_0,\ldots, X_{d-1}] \f$ and a relation \f$ F \f$
+which is a polynomial in \f$ N \f$ variables, we use Sumcheck over the polynomial
+ * \f{align}{
+    \tilde{F}
+    (X_0,\ldots, X_{d-1}) =
+    pow_{\beta}(X_0,\ldots, X_{d-1}) \cdot F\left( P_1 (X_0,\ldots, X_{d-1}), \ldots, P_N (X_0,\ldots, X_{d-1}) \right)
+    \f}
+to establish that \f$ F(P_1(\vec \ell),\ldots, P_N(\vec \ell) ) = 0 \f$, i.e. that \f$ F \f$ is satisfied, at every
+point of \f$\{0,1\}^d\f$.
+
+ In the implementation, the relation polynomial \f$ F \f$ is determined by \p Flavor::Relations which is fed to \ref
+bb::SumcheckProverRound "Sumcheck Round Prover".
+
```

Comment:
General comment on the two sections above: it is nice that they are kept succint but can you work a bit on making them more flowy?

Reply:
Does it still apply with an expanded sumcheck-outline.md?